source/slang/slang-ir-diff-jvp.cpp

```diff
@@ -366,6 +390,77 @@ struct JVPTranscriber
         return nullptr;
     }
 
+    IRInst* differentiateSwizzle(IRBuilder* builder, IRSwizzle* swizzleP)
+    {
+        if (auto baseD = getDifferentialInst(swizzleP->getBase(), nullptr))
+        {
+            List<IRInst*> swizzleIndices;
+
+            // TODO(sai): Is this copy really needed?
```

Comment:
You can just reuse the array in the original inst if it is not changed.
The indices you passed to IRBuilder::emitSwizzle() will be copied into the generated inst anyways.